framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2257,19 +2257,32 @@ public ParameterizedExecutableType constructorFromUse(NewClassTree tree) {
         AnnotatedTypeMirror type = fromNewClass(tree);
         addComputedTypeAnnotations(tree, type);
         AnnotatedExecutableType con = getAnnotatedType(ctor); // get unsubstituted type
+
+        if (tree.getClassBody() != null) {
+            // If this is an anonymous class, replace the parameter types and return type with those
+            // of the super constructor
+            ExecutableElement superCtor = TreeUtils.anonymousSuperConstructor(tree);
+            AnnotatedExecutableType superCtorType = getAnnotatedType(superCtor);
+
+            if (tree.getArguments().size() == superCtorType.getParameterTypes().size() + 1) {
+                // In JDK 8, an anonymous class instantiation with enclosing expression passes
+                // the enclosing expression as the first argument (i.e. synthetic argument),
+                // while the super constructor parameters does not have the counterpart.
+                // Therefore, we get the first parameter of the anonymous constructor and manually
+                // add it to the parameter list at index 0.
+                List<AnnotatedTypeMirror> actualParams = new ArrayList<>();
+                actualParams.add(con.getParameterTypes().get(0));
+                actualParams.addAll(superCtorType.getParameterTypes());
+                superCtorType.setParameterTypes(actualParams);
+            }
+            con = superCtorType;
```

Comment:
Could you add a comment here why this now uses the anonymous super constructor?
I like the separation into `TreeUtils.constructor` and `TreeUtils.anonymousSuperConstructor`, but having a comment here why we switch the Element would be good.
Was it because the synthetic anonymous constructor does not have all the type annotations that the real super constructor has?

Reply:
Yes, that's the main reason. Actually we need to get the annotations on both the return type and the parameter types from the super constructor, if any of them exist. And these are pretty much all about a constructor invocation. So maybe just simply use the super constructor element.